### PR TITLE
feat: allow passing options to auth.getClient

### PR DESCRIPTION
**Pull request by JustinBeckwith**
This change lets us stop pushing people towards createScopedRequired, while not breaking the API for those already using it.  It will also really simplify the getApplicationDefault scenario for users.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
     "opn": "^5.2.0",
     "pify": "^3.0.0",
     "prettier": "^1.11.1",
-    "sinon": "^5.0.0",
+    "sinon": "^4.5.0",
```
Comment (ofrobots) [inline package.json]: Downgrade? why?

File: test/test.googleauth.ts [hunk 2]
```diff
@@ -984,6 +984,7 @@ it('getApplicationDefault should use GCE when well-known file and env const are
          'compute-placeholder', res.credential.credentials.refresh_token);
    });
 
+
```
Comment (ofrobots) [inline test/test.googleauth.ts]: nit: unnecessary extra line.

File: package.json (hunk 1 above)
Comment (JustinBeckwith) [reply package.json]: If you install 5.0, they say it's deprecated.  I checked `npm info sinon`, and it looks like they accidentally released it at one point, then quickly switched back the dist tag to 4.5. I could pull this out to another PR if you'd like :)

File: test/test.googleauth.ts (hunk 2 above)
Comment (JustinBeckwith) [reply test/test.googleauth.ts]: fixed

File: package.json (hunk 1 above)
Comment (ofrobots) [reply package.json]: Thanks for explaining. Sounds fine here.

**Pull request merged**